File: 0033-search-in-rotated-sorted-array/0033-search-in-rotated-sorted-array.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {

        int low =0, high = nums.size()-1;
        while (low <= high) {
           int mid = (low+high)/2;
            if (nums[mid] == target) return mid;
            // left sorted array
            if (nums[mid]>= nums[low]){
                if (nums[mid]>target and target >=nums[low]) high = mid-1;
                else low = mid+1;
            }
            // right sorted array
            else{ 
                if (nums[mid]<target and target <= nums[high]) low = mid+1;
                else high = mid-1;
            }
            cout << mid<< ' ';
        }
        
        return -1;
    }
};

File: 0033-search-in-rotated-sorted-array/0033-search-in-rotated-sorted-array.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l =0
        r = len(nums)-1

        while l<=r:
            mid = (l+r)//2
            if nums[mid] == target:
                return mid
            
            #left sorted array
            if nums[l] <= nums[mid]:
                if target> nums[mid] or target < nums[l]:
                    l = mid+1
                else:
                    r = mid-1
            
            #right sorted array
            else:
                if target < nums[mid] or target > nums[r]:
                    r = mid-1
                else:
                    l = mid+1
        return -1